### ISSUE TITLE
Is there a way to do conditional logging?

### ISSUE THREAD

**Issue by ruipacheco** [gabime/spdlog#388]
I'm currently writing code that looks like this:

```
if (response.size() > 16777215) {
          logger_->warn("...");
}
```

Is there a way to bypass the if statement and just do 
```
logger_->warn((response.size() > 16777215), "...");
```

If not, can this be turned into a feature request? I could write a wrapper around spdlog but I wonder if implementing this within the logger by someone who knows the code will result in faster and more stable code.

**Comment by gabime** [comment]
something like ```log_if(condition, log_level, msg, args)``` ?

**Comment by ruipacheco** [comment]
Very much like Google Log, yes. The condition works of course, but moving that logic to the log makes the condition clearer.

**Comment by gabime** [comment]
merged in #460
